go/plugins/dotprompt/picoschema_test.go

```diff
@@ -97,3 +100,54 @@ func replaceEmptySchemas(m map[string]any) any {
 	}
 	return m
 }
+
+func replaceAnyOfWithTypeArray(schema map[string]any) {
```

Comment:
Can you help me understand what this is for? Is it only needed for the test or everywhere?

Reply:
that it is a helper function, to match exactly the test cases with the string expected, without it we have error because the serializate func outputs something like `string("red", "blue")` and we want `AnyOf("red", "blue")`, but the important refactor I think it is about: picoschema.go https://github.com/firebase/genkit/pull/1400/files#diff-d643807a9a75f3e4805fd792bbd253ad99819a506fcfb5bdf9ffa20660d87a4eR109

Reply:
Can you add a sample with an actual dotprompt file that uses these features that weren't supported before? The tests are good but we're doing quite a bit of extra logic here so I want to make sure that it actually works in practice.